core/src/main/java/org/elasticsearch/rest/action/search/RestMultiSearchAction.java

```diff
@@ -173,6 +181,13 @@ public static void parseMultiLineRequest(RestRequest request, IndicesOptions ind
                             searchRequest.preference(nodeStringValue(value, null));
                         } else if ("routing".equals(entry.getKey())) {
                             searchRequest.routing(nodeStringValue(value, null));
+                        } else if ("timeout".equals(entry.getKey())) {
+                            if (timeout != null) {
```

Comment:
We now have 3 places we can express a timeout - URL param, Action line and Search body - feels like 2 too many to me but I'm not familiar with the reasoning for wanting all these.

Either way, this code only parses and sets an Action-supplied setting if there is also a URL-param supplied setting which is surely wrong?

It think this PR needs some tests to catch stuff like this and to demonstrate the correct support for parsing the various time-formats.